Sources/BCRunner/BytecodeReaders.swift

```diff
@@ -4,78 +4,84 @@ import Foundation
 import BCCommon
 
 protocol BytecodeElement {
-  static func readFrom(_ runner: BytecodeRunner) -> Self
+  static func readFrom(_ runner: BytecodeRunner) throws -> Self
 }
 
+struct EnumerationError<T>: Swift.Error {}
```

Comment:
It’s good to provide some some additional context, like rawValue. Also, EnumerationError is not very good name, maybe something like InvalidRawValue?